gcc/rust/ast/rust-ast.h

```diff
@@ -634,7 +634,7 @@ struct Attribute
 
   location_t locus;
 
-  bool inner_attribute;
+  bool inner_attribute = false;
```

Comment:
I don't think this is required, since the full constructor of that class has a default parameter for `inner_attribute` which sets the field to `false`

Reply:
Yea, you are correct here, I think I got spooked looking at -O2 code in GDB

Reply:
Nevermind, I undid this change, and now I get an ICE because inner_attribute is set to true on an outer attribute.

Reply:
```c++
diff --git a/gcc/rust/ast/rust-ast.cc b/gcc/rust/ast/rust-ast.cc
index 9c0b1bf8245..f4732bfd1fd 100644
--- a/gcc/rust/ast/rust-ast.cc
+++ b/gcc/rust/ast/rust-ast.cc
@@ -311,7 +311,7 @@ Attribute::get_traits_to_derive ()

 // Copy constructor must deep copy attr_input as unique pointer
 Attribute::Attribute (Attribute const &other)
-  : path (other.path), locus (other.locus)
+  : path (other.path), locus (other.locus), inner_attribute (other.inner_attribute)
 {
   // guard to protect from null pointer dereference
   if (other.attr_input != nullptr)
@@ -324,6 +324,7 @@ Attribute::operator= (Attribute const &other)
 {
   path = other.path;
   locus = other.locus;
+  inner_attribute = other.inner_attribute;
   // guard to protect from null pointer dereference
   if (other.attr_input != nullptr)
     attr_input = other.attr_input->clone_attr_input ();
     ```

Reply:
ah yeah didn't catch that... that was indeed missing from the copy constructors. this looks very good and correct, I'd like you to use it yeah :D thanks!

Reply:
I pushed this patch, the testing should be fixed now